src/modules/credentials/services/CredentialService.ts

```diff
@@ -73,11 +75,20 @@ export class CredentialService {
    */
   public async createProposal(
     connectionRecord: ConnectionRecord,
-    config?: Omit<ProposeCredentialMessageOptions, 'id'>
+    config?: CredentialProposeOptions
   ): Promise<CredentialProtocolMsgReturnType<ProposeCredentialMessage>> {
     // Assert
     connectionRecord.assertReady()
 
+    // Add the linked attachments to the credentialProposal
+    if (config?.linkedAttachments) {
+      config.credentialProposal = CredentialUtils.createAndLinkCredentialToAttachment(
```

Comment:
I'd use separate variables instead of modifying the input `config` object. This can result in unexpected behaviour for the framework user